### ISSUE TITLE
Cannot update from v3.2.2 to v3.3.0 due to mimemagic error in bundle install

### ISSUE THREAD

**Issue by clement-analogue** [tootsuite/mastodon#16228]
Hi,

I have a (known) issue while updating from v3.2.2 to v3.3.0.

```
~/live$ bundle install
Fetching gem metadata from https://rubygems.org/............
Your bundle is locked to mimemagic (0.3.5), but that version could not be found in any of the sources listed in your Gemfile. If you
haven't changed sources, that means the author of mimemagic (0.3.5) has removed it. You'll need to update your bundle to a version
other than mimemagic (0.3.5) that hasn't been removed in order to install.
```

I saw this error has been reported for fresh install, but I am updating, so, I don't know what to do from there.

Shall I revert to v3.2.2 with git checkout, or shall I checkout a RC from v3.4.0, or something else?
I am not sure what is the best, and safest, strategy.
If I checkout a RC from v3.4.0, what steps I should perform to upgrade?

Thank you!

**Comment by ClearlyClaire** [comment]
Duplicate of #15986 and #16102, this is fixed in `main` and v3.4.0 release candidates (though there is one known important bug in the latest release candidate, so you're better off using `main` instead).

In order to switch to `main` you should:
1. check out the code (`git fetch && git checkout origin/main`)
2. install dependencies (`bundle install && yarn install`)
3. run the pre-deployment migration scripts (`SKIP_POST_DEPLOYMENT_MIGRATIONS=true RAILS_ENV=production bundle exec rails db:migrate`)
4. precompile the assets (`RAILS_ENV=production bundle exec rails assets:precompile`)
5. restart mastodon processes (`systemctl reload mastodon-web && systemctl restart mastodon-{sidekiq,streaming}`)
6. run the post-deployment migration scripts (`RAILS_ENV=production bundle exec rails db:migrate`)
7. clear Rails' cache (`RAILS_ENV=production bin/tootctl cache clear`)

**Comment by clement-analogue** [comment]
Great, works perfectly. Thank you for the detailed steps!
